packages/a11y-base/src/tabindex-mixin.js

```diff
@@ -70,9 +74,37 @@ export const TabindexMixin = (superclass) =>
      * @protected
      */
     _tabindexChanged(tabindex) {
+      if (this._shouldAllowFocusWhenDisabled()) {
+        return;
+      }
+
       if (this.disabled && tabindex !== -1) {
         this._lastTabIndex = tabindex;
         this.tabindex = -1;
       }
     }
+
+    /**
+     * Overrides the default element `focus` method in order to prevent
+     * focusing the element when it is disabled.
+     *
+     * @protected
+     * @override
+     */
+    focus() {
+      if (!this.disabled || this._shouldAllowFocusWhenDisabled()) {
+        super.focus();
+      }
+    }
```

Comment:
Previously it was possible to programmatically move focus to a `disabled` button. This change affects the behaviour even with the feature flag off. It is intentional?

Reply:
I can extract this change to a separate PR by the way.

Reply:
Should the change also go to `focus-mixin` rather than `tabindex-mixin`?

Reply:
I also had this dilemma. My reasoning for placing this change to TabindexMixin was that `focus()` has to be prevented specifically for components that are made focusable with `tabindex=0`, because otherwise, it works out of the box.

Reply:
Created a separate PR with this change: https://github.com/vaadin/web-components/pull/8540